src/test/java/com/google/sps/DestinationDataServletTest.java

```diff
@@ -0,0 +1,135 @@
+  private static final String LNG_INPUT = "234.567";
+  private static final String CITY_INPUT = "San Francisco";
+  private static final String DESC_INPUT = "Famous Bridge in SF";
+  private static final String RIDDLE_INPUT = "Stay away from me if you're afraid of heights";
+  private static final String H1_INPUT = "Overlooks the water";
+  private static final String H2_INPUT = "Golden-red in color";
+  private static final String H3_INPUT = "You have to pay to use me";
+  private static final String[] TAG_INPUT = {"historical", "tourist"};
+  private static final String OBSCURITY_INPUT = "easy";
+  private static final String PLACEID_INPUT = "123";
+
+  private static final Gson GSON = new Gson();
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void storeInDatastore() throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    servlet = new DestinationDataServlet(datastore);
+    doReturn(NAME_INPUT).when(request).getParameter(DestinationDataServlet.NAME_PARAMETER);
+    doReturn(LAT_INPUT).when(request).getParameter(DestinationDataServlet.LAT_PARAMETER);
+    doReturn(LNG_INPUT).when(request).getParameter(DestinationDataServlet.LNG_PARAMETER);
+    doReturn(CITY_INPUT).when(request).getParameter(DestinationDataServlet.CITY_PARAMETER);
+    doReturn(DESC_INPUT).when(request).getParameter(DestinationDataServlet.DESCRIPTION_PARAMETER);
+    doReturn(RIDDLE_INPUT).when(request).getParameter(DestinationDataServlet.RIDDLE_PARAMETER);
+    doReturn(H1_INPUT).when(request).getParameter(DestinationDataServlet.HINT1_PARAMETER);
+    doReturn(H2_INPUT).when(request).getParameter(DestinationDataServlet.HINT2_PARAMETER);
+    doReturn(H3_INPUT).when(request).getParameter(DestinationDataServlet.HINT3_PARAMETER);
+    doReturn(TAG_INPUT).when(request).getParameterValues(DestinationDataServlet.TAG_PARAMETER);
+    doReturn(OBSCURITY_INPUT)
+        .when(request)
+        .getParameter(DestinationDataServlet.OBSCURITY_PARAMETER);
+    doReturn(PLACEID_INPUT).when(request).getParameter(DestinationDataServlet.PLACEID_PARAMETER);
+    servlet.doPost(request, response);
+
+    Query query = new Query(Constants.DESTINATION_ENTITY);
+    PreparedQuery results = datastore.prepare(query);
+
+    Assert.assertEquals(1, results.countEntities());
+
+    Destination destination =
+        GSON.fromJson(
+            (String) results.asSingleEntity().getProperty(Constants.DESTINATION_JSON),
+            Destination.class);
+
+    String actual = GSON.toJson(destination);
+
+    String expected = getExpectedDestination();
+
+    Assert.assertEquals(actual, expected);
+
+    verify(response).sendRedirect(DestinationDataServlet.HOME_URL);
```

Comment:
Tests should be structured in three distinct chunks: 1) arrange 2) act 3) assert.  This gives the reader a visual clue about what the test code is doing.

Therefore, you should rearrange your code like this:
// Arrange
DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
servlet = new DestinationDataServlet(datastore);
doReturn....

// Act
servlet.doPost(request, response);

// Assert
PreparedQuery results = ...
String actual = ...
Assert.assertEquals(1, results.countEntities());
Assert.assertEquals(actual, getExpectedDestination());